dask/highlevelgraph.py

```diff
@@ -153,6 +153,9 @@ def items(self):
     def __iter__(self):
         return toolz.unique(toolz.concat(self.layers.values()))
 
+    def values(self):
+        return toolz.concat(layer.values() for layer in self.layers.values())
```

Comment:
This seems fine to me in practice, but does create a structure different than a `dict_values` view, which means that it has different properties. We should at least return a list instead of a generator, but I'm not sure if there's a better structure to use.

Reply:
We should probably also do this for `keys` and `items`?

Reply:
Perhaps we can just implement `items` and have the others come for free?

Why a list rather than a generator?

Reply:
Because a generator doesn't fullfill the contract of `values`/`keys`/items` (e.g. `item in d.values()`). That's why Python 2 also had `itervalues`/`iterkeys`, `iteritems`.

Reply:
@jcrist am I missing something? It looks like `item in d.values()` is fullfilled

```python
In [1]: import dask.array as da                                                                                             

In [2]: a = da.ones(10, chunks=2)                                                                                           

In [3]: hlg = a.dask                                                                                                        

In [4]: list(hlg.values())[0] in hlg.values()                                                                               
Out[4]: True
```

Reply:
@mrocklin to your point, the current `items` method already avoids the `HighLevelGraph.__getitem__()` method, so could be used as is to build the `keys` and `values` logic. E.g. something like

```python
def values(self):
    for _, val in self.items():
        yield val
```

Reply:
Looks like `itertools.chain` implements `__contains__`, but it's implemented by consuming the generator until a match is found, leading to poor performance and odd behavior:

```python
In [34]: res = toolz.concat([[1, 2, 3], [4], [5, 6, 7]])

In [35]: 1 in res
Out[35]: True

In [36]: 1 in res
Out[36]: False
```